tests/Dockerfile.hepcrawl_base

```diff
@@ -7,36 +7,23 @@
 # under the terms of the Revised BSD License; see LICENSE file for
 # more details.
 
-FROM centos:7
+FROM python:2.7
 
-RUN yum install -y epel-release && \
-    yum update -y && \
-    yum install -y \
-        file \
-        gcc \
-        git \
-        libffi-devel \
-        libxml2-devel \
-        libxslt-devel \
-        libssl-devel \
-        make \
-        openssl-devel \
-        poppler-utils \
-        python-pip \
-        python-virtualenv && \
-    yum clean all
+ARG GID
+ARG UID
+ENV PYTHONDONTWRITEBYTECODE=1
+RUN mkdir /code /var/lib/scrapy /venv
+RUN groupadd test -r -f -g $GID
+RUN useradd test -u $UID -g $GID -d /home/test -m
 
-RUN mkdir /code /hepcrawl_venv /var/lib/scrapy
-
-RUN useradd test
-RUN chown -R test:test /code /hepcrawl_venv /var/lib/scrapy
-
-ADD ./docker_entrypoint.sh /docker_entrypoint.sh
-ADD ./fix_rights /fix_rights
-RUN chmod 4755 /fix_rights
+copy . /code
+RUN chown -R test:test /var/lib/scrapy /venv /code
 
 USER test
+ENV PATH="/home/test/.local/bin:${PATH}"
+
 WORKDIR /code
+RUN pip install --upgrade wheel setuptools idutils rfc3987 bleach jsonschema inspire-utils
```

Comment:
why do you need to install some of the deps explicitly?

Reply:
Because somehow they are not installed. I'm not sure why actually, I can investigate it more if you prefer.

Reply:
Ok, I look into it a little bit more and looks like inspire-schemas are missing this packages in [`setup_requires`](https://github.com/inspirehep/inspire-schemas/blob/ab35f92f667cea0974cb54f717ef6913ba30cf65/setup.py#L231) directive. Without installing them manually build for inspire-schemas is failing:
```bash
  ERROR: Command errored out with exit status 1:
   command: /usr/local/bin/python -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-zHeIhV/inspire-schemas/setup.py'"'"'; __file__='"'"'/tmp/pip-install-zHeIhV/inspire-schemas/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' bdist_wheel -d /tmp/pip-wheel-0XIc4V
       cwd: /tmp/pip-install-zHeIhV/inspire-schemas/
  Complete output (103 lines):
  /usr/local/lib/python2.7/distutils/dist.py:267: UserWarning: Unknown distribution option: 'build_requires'
    warnings.warn(msg)
  /home/test/.local/lib/python2.7/site-packages/autosemver/__init__.py:156: UserWarning: Using explicit parametes is deprecated and will be removed, pass a dictionary as the value of the autosemver parameter instead.
    'Using explicit parametes is deprecated and will be removed, '
  /home/test/.local/lib/python2.7/site-packages/setuptools/dist.py:476: UserWarning: Normalizing '61.3.9
  ' to '61.3.9'
    normalized_version,
  running bdist_wheel
  running build
  running build_py
  creating build
  creating build/lib
  creating build/lib/inspire_schemas
  copying inspire_schemas/utils.py -> build/lib/inspire_schemas
  copying inspire_schemas/errors.py -> build/lib/inspire_schemas
  copying inspire_schemas/__init__.py -> build/lib/inspire_schemas
  copying inspire_schemas/api.py -> build/lib/inspire_schemas
  creating build/lib/inspire_schemas/records
  copying inspire_schemas/records/__init__.py -> build/lib/inspire_schemas/records
  creating build/lib/inspire_schemas/countries
  copying inspire_schemas/countries/__init__.py -> build/lib/inspire_schemas/countries
  creating build/lib/inspire_schemas/builders
  copying inspire_schemas/builders/references.py -> build/lib/inspire_schemas/builders
  copying inspire_schemas/builders/seminars.py -> build/lib/inspire_schemas/builders
  copying inspire_schemas/builders/conferences.py -> build/lib/inspire_schemas/builders
  copying inspire_schemas/builders/__init__.py -> build/lib/inspire_schemas/builders
  copying inspire_schemas/builders/authors.py -> build/lib/inspire_schemas/builders
  copying inspire_schemas/builders/jobs.py -> build/lib/inspire_schemas/builders
  copying inspire_schemas/builders/signatures.py -> build/lib/inspire_schemas/builders
  copying inspire_schemas/builders/literature.py -> build/lib/inspire_schemas/builders
  copying inspire_schemas/builders/builder.py -> build/lib/inspire_schemas/builders
  creating build/lib/inspire_schemas/readers
  copying inspire_schemas/readers/conference.py -> build/lib/inspire_schemas/readers
  copying inspire_schemas/readers/__init__.py -> build/lib/inspire_schemas/readers
  copying inspire_schemas/readers/literature.py -> build/lib/inspire_schemas/readers
  creating build/lib/inspire_schemas/records/elements
  copying inspire_schemas/records/elements/__init__.py -> build/lib/inspire_schemas/records/elements
  copying inspire_schemas/records/institutions.json -> build/lib/inspire_schemas/records
  copying inspire_schemas/records/jobs.json -> build/lib/inspire_schemas/records
  copying inspire_schemas/records/seminars.json -> build/lib/inspire_schemas/records
  copying inspire_schemas/records/experiments.json -> build/lib/inspire_schemas/records
  copying inspire_schemas/records/hep.json -> build/lib/inspire_schemas/records
  copying inspire_schemas/records/journals.json -> build/lib/inspire_schemas/records
  copying inspire_schemas/records/data.json -> build/lib/inspire_schemas/records
  copying inspire_schemas/records/authors.json -> build/lib/inspire_schemas/records
  copying inspire_schemas/records/conferences.json -> build/lib/inspire_schemas/records
  copying inspire_schemas/countries/iso_3166-3.json -> build/lib/inspire_schemas/countries
  copying inspire_schemas/countries/iso_3166-1.json -> build/lib/inspire_schemas/countries
  copying inspire_schemas/records/elements/address.json -> build/lib/inspire_schemas/records/elements
  copying inspire_schemas/records/elements/reference.json -> build/lib/inspire_schemas/records/elements
  copying inspire_schemas/records/elements/acquisition_source.json -> build/lib/inspire_schemas/records/elements
  copying inspire_schemas/records/elements/cnum.json -> build/lib/inspire_schemas/records/elements
  copying inspire_schemas/records/elements/source.json -> build/lib/inspire_schemas/records/elements
  copying inspire_schemas/records/elements/contact.json -> build/lib/inspire_schemas/records/elements
  copying inspire_schemas/records/elements/inspire_field.json -> build/lib/inspire_schemas/records/elements
  copying inspire_schemas/records/elements/id.json -> build/lib/inspire_schemas/records/elements
  copying inspire_schemas/records/elements/records-files.json -> build/lib/inspire_schemas/records/elements
  copying inspire_schemas/records/elements/url.json -> build/lib/inspire_schemas/records/elements
  copying inspire_schemas/records/elements/arxiv_categories.json -> build/lib/inspire_schemas/records/elements
  copying inspire_schemas/records/elements/inspire_experiment_classification.json -> build/lib/inspire_schemas/records/elements
  copying inspire_schemas/records/elements/degree_type.json -> build/lib/inspire_schemas/records/elements
  copying inspire_schemas/records/elements/sourced_value.json -> build/lib/inspire_schemas/records/elements
  copying inspire_schemas/records/elements/related_record.json -> build/lib/inspire_schemas/records/elements
  copying inspire_schemas/records/elements/country_code.json -> build/lib/inspire_schemas/records/elements
  copying inspire_schemas/records/elements/title.json -> build/lib/inspire_schemas/records/elements
  copying inspire_schemas/records/elements/document_type.json -> build/lib/inspire_schemas/records/elements
  copying inspire_schemas/records/elements/json_reference.json -> build/lib/inspire_schemas/records/elements
  copying inspire_schemas/records/elements/material.json -> build/lib/inspire_schemas/records/elements
  copying inspire_schemas/records/elements/language_code.json -> build/lib/inspire_schemas/records/elements
  copying inspire_schemas/records/elements/rank.json -> build/lib/inspire_schemas/records/elements
  warning: build_py: byte-compiling is disabled, skipping.
  
  installing to build/bdist.linux-x86_64/wheel
  running install
  Traceback (most recent call last):
    File "<string>", line 1, in <module>
    File "/tmp/pip-install-zHeIhV/inspire-schemas/setup.py", line 240, in <module>
      do_setup()
    File "/tmp/pip-install-zHeIhV/inspire-schemas/setup.py", line 235, in do_setup
      autosemver=True,
    File "/home/test/.local/lib/python2.7/site-packages/setuptools/__init__.py", line 162, in setup
      return distutils.core.setup(**attrs)
    File "/usr/local/lib/python2.7/distutils/core.py", line 151, in setup
      dist.run_commands()
    File "/usr/local/lib/python2.7/distutils/dist.py", line 953, in run_commands
      self.run_command(cmd)
    File "/usr/local/lib/python2.7/distutils/dist.py", line 972, in run_command
      cmd_obj.run()
    File "/home/test/.local/lib/python2.7/site-packages/wheel/bdist_wheel.py", line 326, in run
      self.run_command('install')
    File "/usr/local/lib/python2.7/distutils/cmd.py", line 326, in run_command
      self.distribution.run_command(command)
    File "/usr/local/lib/python2.7/distutils/dist.py", line 972, in run_command
      cmd_obj.run()
    File "/tmp/pip-install-zHeIhV/inspire-schemas/setup.py", line 50, in run
      _generate_json_schemas()
    File "/tmp/pip-install-zHeIhV/inspire-schemas/setup.py", line 160, in _generate_json_schemas
      _generate_country_code(schemas_dir)
    File "/tmp/pip-install-zHeIhV/inspire-schemas/setup.py", line 139, in _generate_country_code
      from inspire_schemas.utils import COUNTRY_CODE_TO_NAME
    File "inspire_schemas/utils.py", line 34, in <module>
      import rfc3987
  ImportError: No module named rfc3987
  ----------------------------------------

```
It's not preventing inspire-schemas to install at the end, as this is retried after other packages are installed, but it slows down image build a little. I can remove them if you want.